x/move/keeper/contracts/sources/Vesting.move

```diff
@@ -0,0 +1,310 @@
+        let store = borrow_global<VestingStore>(creator);
+        store.token_metadata
+    }
+
+    #[view]
+    /// Returns the address of the vesting store. This address can be used
+    /// to deposit vesting funds.
+    public fun store_addr(creator: address): address acquires VestingStore {
+        let store = borrow_global<VestingStore>(creator);
+        object::address_from_extend_ref(&store.extend_ref)
+    }
+
+    #[view]
+    public fun vesting_funds(creator: address): u64 acquires VestingStore {
+        let store = borrow_global<VestingStore>(creator);
+        let store_addr = object::address_from_extend_ref(&store.extend_ref);
+        primary_fungible_store::balance(store_addr, store.token_metadata)
+    }
+
+    #[view]
+    public fun has_vesting(creator: address, recipient: address): bool acquires VestingStore {
+        let store = borrow_global<VestingStore>(creator);
+
+        table::contains(&store.vestings, recipient)
+    }
+
+    #[view]
+    public fun vesting_info(creator: address, recipient: address): Vesting acquires VestingStore {
+        let store = borrow_global<VestingStore>(creator);
+        assert!(
+            table::contains(&store.vestings, recipient),
+            error::not_found(EVESTING_NOT_EXISTS)
+        );
+
+        *table::borrow(&store.vestings, recipient)
+    }
+
+    #[view]
+    public fun claimable_amount(creator: address, recipient: address): u64 acquires VestingStore {
+        let store = borrow_global<VestingStore>(creator);
+        assert!(
+            table::contains(&store.vestings, recipient),
+            error::not_found(EVESTING_NOT_EXISTS)
+        );
+
+        let vesting = table::borrow(&store.vestings, recipient);
+        calc_claimable_amount(vesting)
+    }
+
+    #[view]
+    public fun vesting_amount(creator: address, recipient: address): u64 acquires VestingStore {
+        let store = borrow_global<VestingStore>(creator);
+        assert!(
+            table::contains(&store.vestings, recipient),
+            error::not_found(EVESTING_NOT_EXISTS)
+        );
+
+        let vesting = table::borrow(&store.vestings, recipient);
+        vesting.allocation - calc_claimable_amount(vesting)
+    }
+
+    #[view]
+    public fun vesting_table_handle(creator: address): address acquires VestingStore {
+        let store = borrow_global<VestingStore>(creator);
+        table::handle(&store.vestings)
+    }
+
+    // Internal functions
+
+    fun calc_claimable_amount(vesting: &Vesting): u64 {
+        let (_, cur_time) = block::get_block_info();
+        let cliff_time = vesting.start_time + vesting.cliff_period;
+
+        // check if the vesting is still in the cliff period
+        if (cur_time < cliff_time) {
+            return 0
+        };
+
+        // calculate elapsed claim frequencies
+        let elapsed_claim_frequencies = (cur_time - cliff_time) / vesting.claim_frequency;
```

Comment:
Is there a case where total allocation amount is received at once after cliff time has passed? If so, we need to consider the case where the frequency is 0.